osu.Game/Online/Chat/MessageFormatter.cs

```diff
@@ -28,7 +28,7 @@ public static class MessageFormatter
         //      http[s]://<domain>.<tld>[:port][/path][?query][#fragment]
         private static readonly Regex advanced_link_regex = new Regex(
             // protocol
-            @"(?<link>[a-z]*?:\/\/" +
+            @"(?<link>(https?|osu)[a-z]*?:\/\/" +
```

Comment:
This still lets through

```
httpwhatever://
osuicanputabsolutelyanythinghere://
```

![1694713603](https://github.com/ppy/osu/assets/20418176/792d153f-3164-417f-b7c3-38e5e4317c19)

I'm _guessing_ this was trying to support `osump`, but if you're gonna allowlist, then allowlist properly, by being extremely specific:

```suggestion
            @"(?<link>(https?|osu(mp)?):\/\/" +
```

This should have test coverage, too, by the way.